### ISSUE TITLE
readOnly schema property wrong behaviour

### ISSUE THREAD

**Issue by Adam-Michalski** [ngx-formly/ngx-formly#1566]
bug report

**Current behavior**
If `readOnly` is present in schema field is disabled no matter what value it has.

**Expected behavior**
`readOnly` value should represent state (disabled/enabled) of form control.

[`If "readOnly" has a value of boolean true, it indicates that the value of the instance is managed exclusively by the owning authority, and attempts by an application to modify the value of this property are expected to be ignored or rejected by that owning authority.`](https://json-schema.org/latest/json-schema-validation.html#rfc.section.10.3)

**Minimal reproduction of the problem**
[StackBlitz](https://stackblitz.com/edit/angular-q5jhl7)

It blocks different implementations even with mapFunction

```  
public mapFun(mappedField: FormlyFieldConfig, mapSource: JSONSchema7 & MySchema): FormlyFieldConfig {
    mappedField.hideExpression = mapSource.hidden;
    mappedField.templateOptions.disabled = false; //after this line field with readOnly property is still disabled
    return mappedField;
}
```

**Comment by aitboudad** [comment]
`readOnly` should be boolean, if you want to use disabled state pass it throughout `widget` or use `map` fn, 
here is a working example https://stackblitz.com/edit/angular-q5jhl7-atef96

**Comment by Adam-Michalski** [comment]
It doesn't matter what is in `readOnly` property if it is present I cannot change disabled state. Field is always disabled. Cannot change it trought widget. How is this the way it should be. If you can present the way to change this by checkbox or something cause your described way doesn't work:
[example](https://stackblitz.com/edit/angular-q5jhl7-g1vrej?file=src/app/app.component.ts)

**Comment by aitboudad** [comment]
fixed in `5.1.2`, passing `false` value to `readOnly` should not disable the input.
